packages/api-derive/src/accounts/accountId.ts

```diff
@@ -24,7 +24,7 @@ export function accountId (instanceId: string, api: DeriveApi): (address?: Addre
       : decodeAddress((address || '').toString());
 
     if (decoded.length > 8) {
-      return of(api.registry.createType('AccountId', decoded));
+      return of(api.registry.createType(decoded.length === 20 ? 'AccountId20' : 'AccountId', decoded));
```

Comment:
I wonder if this would be good to use `encodedLength` as well.

Reply:
Ahhh this actually doesn't make sense since `encodedLength` is only available for Codec classes